app/controllers/katello/api/v2/host_subscriptions_controller.rb

```diff
@@ -0,0 +1,88 @@
+module Katello
+  class Api::V2::HostSubscriptionsController < Katello::Api::V2::ApiController
+    before_filter :find_host
+    before_filter :check_subscriptions, :only => [:add_subscriptions, :remove_subscriptions]
+
+    resource_description do
+      api_version 'v2'
+      api_base_url "/api"
+    end
+
+    api :GET, "/hosts/:host_id/subscriptions", N_("List a host's subscriptions")
+    param :host_id, Integer, :desc => N_("Id of the host"), :required => true
+    def index
+      @collection = index_response
+      respond_for_index :collection => @collection
+    end
+
+    def index_response
+      entitlements = @host.subscription_facet.candlepin_consumer.entitlements
+      subscriptions = entitlements.map { |entitlement| ::Katello::HostSubscriptionPresenter.new(entitlement) }
+      full_result_response(subscriptions)
+    end
+
+    api :PUT, "/hosts/:host_id/subscriptions/auto_attach", N_("Trigger an auto-attach of subscriptions")
+    param :host_id, Integer, :desc => N_("Id of the host"), :required => true
+    def auto_attach
+      sync_task(::Actions::Katello::Host::AutoAttachSubscriptions, @host)
+      respond_for_index(:collection => index_response, :template => "index")
+    end
+
+    api :GET, "/hosts/:host_id/subscriptions/events", N_("List subscription events for the host")
+    param :host_id, Integer, :desc => N_("Id of the host"), :required => true
+    def events
+      respond_for_index :collection => { :results => @host.subscription_facet.candlepin_consumer.events.map { |e| OpenStruct.new(e) } }
```

Comment:
I clicked all of those things and still had no events showing up, are those events dependent on the message bus being stable?